openedx/core/djangoapps/credentials/utils.py

```diff
@@ -0,0 +1,88 @@
+"""Helper functions for working with Credentials."""
+from __future__ import unicode_literals
+import logging
+
+from edx_rest_api_client.client import EdxRestApiClient
+
+from openedx.core.djangoapps.credentials.models import CredentialsApiConfig
+from openedx.core.djangoapps.programs.utils import get_programs_for_credentials
+from openedx.core.lib.token_utils import get_id_token
+
+
+log = logging.getLogger(__name__)
+
+
+def get_user_credentials(user):
+    """Given a user, get credentials earned from the Credentials service.
+
+    Arguments:
+        user (User): The user to authenticate as when requesting credentials.
+
+    Returns:
+        list of dict, representing credentials returned by the Credentials
+        service.
+    """
+    credentials_config = CredentialsApiConfig.current()
+    credentials = []
+    if not credentials_config.enabled:
+        log.warning('Credentials configuration is disabled.')
+        return credentials
+
+    try:
+        jwt = get_id_token(user, credentials_config.OAUTH2_CLIENT_NAME)
+        api = EdxRestApiClient(credentials_config.internal_api_url, jwt=jwt)
+    except Exception:  # pylint: disable=broad-except
+        log.exception('Failed to initialize the Credentials API client.')
+        return credentials
+
+    try:
+        response = api.user_credentials.get(username=user.username)
+        credentials = response.get('results')
+        page = 1
+        next_page = response.get('next', None)
+        while next_page:
+            page += 1
+            response = api.user_credentials.get(page=page, username=user.username)
+            credentials = credentials + response.get('results')
+            next_page = response.get('next', None)
+
+    except Exception:  # pylint: disable=broad-except
+        log.exception('Failed to retrieve programs from the Programs API.')
```

Comment:
It looks like you copied a lot of the code here from the implementation of `get_programs` in the programs utils. This comment wasn't updated. This code isn't retrieving programs from the Programs API, it's retrieving credentials from the Credentials API.

You should reconsider duplicating this much code. The Programs and Credentials APIs are very similar. Both APIs include `next` attributes in their responses. Caching may only be appropriate for the Programs API at the moment, but it's use is governed by a configuration model. Can you use the same code to make requests against both?